src/Facility.Definition/ServiceFieldValidation.cs

```diff
@@ -0,0 +1,123 @@
+using System;
+using System.Globalization;
+
+namespace Facility.Definition
+{
+	/// <summary>
+	/// A validation criterion of a DTO field.
+	/// </summary>
+	public sealed class ServiceFieldValidation
+	{
+		/// <summary>
+		/// Creates a validation criterion of a DTO field.
+		/// </summary>
+		internal ServiceFieldValidation(ServiceAttributeInfo attribute)
+		{
+			Attribute = attribute ?? throw new ArgumentNullException(nameof(attribute));
+
+			foreach (var parameterInfo in attribute.Parameters)
+			{
+				switch (parameterInfo.Name)
+				{
+					case "length":
+						LengthRange = GetRange<ulong>(attribute, parameterInfo, s => ulong.TryParse(s, NumberStyles.Number, CultureInfo.InvariantCulture, out var value) ? value : null);
+						break;
+					case "count":
+						CountRange = GetRange<ulong>(attribute, parameterInfo, s => ulong.TryParse(s, NumberStyles.Number, CultureInfo.InvariantCulture, out var value) ? value : null);
+						break;
+					case "value":
+						ValueRange = GetRange<long>(attribute, parameterInfo, s => long.TryParse(s, NumberStyles.Number, CultureInfo.InvariantCulture, out var value) ? value : null);
+						break;
+					case "regex":
+						RegexPattern = parameterInfo.Value;
+						break;
+					default:
+						parameterInfo.AddValidationError(ServiceDefinitionUtility.CreateUnexpectedAttributeParameterError(parameterInfo.Name, parameterInfo));
+						break;
+				}
+			}
+		}
+
+		internal ServiceAttributeInfo Attribute { get; }
+
+		/// <summary>
+		/// Allowed range for the collection entry count.
+		/// </summary>
+		public ServiceFieldValidationRange<ulong>? CountRange { get; }
```

Comment:
```suggestion
		public ServiceFieldValidationRange<int>? CountRange { get; }
```

Unsigned integers are rarely used in .NET APIs, and `int.MaxValue` is more than enough for a collection count in this context. Same with `LengthRange` below.